### PR TITLE
[T3A4][T16-A1]Lai Zhiyuan

### PR DESCRIPTION
**Pull request by zyjames**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 public class Main {
 
     /** Version info of the program. */
-    public static final String VERSION = "AddessBook Level 2 - Version 1.0";
+    public static final String VERSION = "AddessBook Level 2 - Version 1.2";
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java]: this is a bit of an overkill :-P

File: src/seedu/addressbook/Main.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ private void start(String[] launchArgs) {
             this.addressBook = storage.load();
             ui.showWelcomeMessage(VERSION, storage.getPath());
 
-        } catch (InvalidStorageFilePathException | StorageOperationException e) {
+        } catch (InvalidStorageFilePathException | StorageOperationException | FileNotFoundException e) {
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java]: this need not be only on `init`. the file can go missing anytime during the program execution.
hence, i would suggest a better way of handling the file not found exception here.

File: src/seedu/addressbook/storage/StorageFile.java [hunk 3]
```diff
@@ -85,29 +93,40 @@ public StorageFile(String filePath) throws InvalidStorageFilePathException {
...
     /**
      * Saves all data to this storage file.
      *
+     * @throws FileNotFoundException if the file does not exist.
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
-
+    public void save(AddressBook addressBook) throws StorageOperationException, FileNotFoundException {
+        if (!isFileExist()) {
```
Comment (okkhoy) [inline src/seedu/addressbook/storage/StorageFile.java]: you have got the basic idea right; however, in the current form, the user experience is degraded.
as soon as the file goes missing, the program halts.

in the original implementation, the file is created on the fly; think how you can retain the behavior while still using exception as a means to inform the user that the file went missing!!

**Comment by okkhoy** [comment]
@zyjames 
Some comments added. Please ack & close the PR after reading comments.

File: src/seedu/addressbook/Main.java (hunk 1 above)
Comment (zyjames) [reply src/seedu/addressbook/Main.java:24]: I thought any change of the code should update the Version of the Address Book?

File: src/seedu/addressbook/storage/StorageFile.java (hunk 3 above)
Comment (zyjames) [reply src/seedu/addressbook/storage/StorageFile.java:113]: Thanks for pointing out my mistake! Will think of a better way to improve the user experience! (:

**Pull request closed without merging**